src/hydra-eval-jobs/hydra-eval-jobs.cc

```diff
@@ -442,45 +442,66 @@ int main(int argc, char * * argv)
         for (auto i = state->jobs.begin(); i != state->jobs.end(); ++i) {
             auto jobName = i.key();
             auto & job = i.value();
+            // For the error message
+            std::string lastTriedJobName = i.key();
```

Comment:
I guess the initialization here is pretty useless...

I think when I wrote that up originally I was thinking about using `auto` to intrinsically get the type. In that case `auto lastTriedJobName = jobName`  is probably better.

Otherwise, initializing with `""`.

What would be the better option?

Reply:
IIRC the nlohman value `key()` function returns reference. `auto` should then deduct the type `T&` which is probably not what we want here. The assignments in line 454 and 466 would then override the value in the JSON document. That being said my type deduction knowledge might be a bit rusty... I would stick with the string there.